test/python/pulse/test_block.py

```diff
@@ -749,3 +750,227 @@ def test_parametrized_context(self):
         ref_sched = ref_sched.insert(90, pulse.Delay(10, self.d0))
 
         self.assertScheduleEqual(block, ref_sched)
+
+
+class TestBlockFilter(BaseTestBlock):
+    """Test ScheduleBlock filtering methods."""
+
+    def test_filter_channels(self):
+        """Test filtering over channels."""
+        # test a non-nested block
+        blk = pulse.ScheduleBlock()
```

Comment:
Could you please write these test object in the builder syntax for better readability?

Reply:
Done! 😃  👉 https://github.com/Qiskit/qiskit-terra/pull/9971/commits/a13945b1886277b85ac82f6c280d2cea96623457